src/test/java/org/opensearch/neuralsearch/processor/RRFProcessorIT.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor;
+
+import lombok.SneakyThrows;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.index.query.MatchQueryBuilder;
+import org.opensearch.neuralsearch.BaseNeuralSearchIT;
+import org.opensearch.neuralsearch.query.HybridQueryBuilder;
+import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.opensearch.neuralsearch.util.TestUtils.DELTA_FOR_SCORE_ASSERTION;
+
+public class RRFProcessorIT extends BaseNeuralSearchIT {
+
+    private int currentDoc = 1;
+    private static final String RRF_INDEX_NAME = "rrf-index";
+
+    @SneakyThrows
+    public void testRRF_whenValidInput_thenSucceed() {
+        String ingestPipelineName = "rrf-ingest-pipeline";
+        String modelId = null;
+        try {
+            modelId = prepareModel();
+            createPipelineProcessor(modelId, ingestPipelineName, ProcessorType.TEXT_EMBEDDING);
+            Settings indexSettings = Settings.builder().put("index.knn", true).put("default_pipeline", ingestPipelineName).build();
+            String indexMappings = XContentFactory.jsonBuilder()
```

Comment:
same should be possible with one variations of existing methods for index creation, e.g. https://github.com/opensearch-project/neural-search/blob/main/src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java#L879. Yuo can avoid most of boilerplate code.